theano/gpuarray/opt.py

```diff
@@ -369,32 +648,33 @@ def local_gpureshape(node, context_name):
 
 @register_opt('fast_compile')
 @op_lifter([tensor.Rebroadcast])
-def local_gpu_rebroadcast(node, context_name):
-    return node.op(as_gpuarray_variable(node.inputs[0], context_name))
+@register_opt2([tensor.Rebroadcast], 'fast_compile')
+def local_gpu_rebroadcast(op, context_name, inputs, outputs):
+    return op(as_gpuarray_variable(inputs[0], context_name))
 
 
 @register_opt('fast_compile')
 @op_lifter([tensor.Flatten])
-def local_gpuflatten(node, context_name):
-    op = node.op
+@register_opt2([tensor.Flatten], 'fast_compile')
+def local_gpuflatten(op, context_name, inputs, outputs):
     shp = []
     if op.outdim != 1:
-        shp = [node.inputs[0].shape[i] for i in range(op.outdim - 1)]
+        shp = [inputs[0].shape[i] for i in range(op.outdim - 1)]
     shp += [-1]
     res = GpuReshape(op.outdim, None)
-    o = res(node.inputs[0], theano.tensor.as_tensor_variable(shp))
+    o = res(inputs[0], theano.tensor.as_tensor_variable(shp))
     return o
 
 
 @register_opt('fast_compile')
 @op_lifter([tensor.Elemwise])
-def local_gpu_elemwise(node, context_name):
-    op = node.op
+@register_opt2([tensor.Elemwise], 'fast_compile')
+def local_gpu_elemwise(op, context_name, inputs, outputs):
```

Comment:
rename to local_gpua_elemwise for consistency. I know, it was already before this PR like that.